SYCL/DeviceLib/built-ins/ext_native_math.cpp

```diff
@@ -0,0 +1,96 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fsycl-device-code-split=per_kernel %s -o %t.out
+// RUN: %HOST_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+// OpenCL CPU driver does not support cl_khr_fp16 extension for this reason this
+// test is compiled with the -fsycl-device-code-split flag
+
+#include <CL/sycl.hpp>
+#include <cassert>
+
+template <typename T> void assert_out_of_bound(T val, T lower, T upper) {
+  assert(sycl::all(lower < val && val < upper));
+}
+
+template <>
+void assert_out_of_bound<float>(float val, float lower, float upper) {
+  assert(lower < val && val < upper);
+}
+
+template <>
+void assert_out_of_bound<sycl::half>(sycl::half val, sycl::half lower,
+                                     sycl::half upper) {
+  assert(lower < val && val < upper);
+}
+
+template <typename T> void native_tanh_tester(sycl::queue q) {
+  T r{0};
+
+#ifdef SYCL_EXT_ONEAPI_NATIVE_MATH
+  {
+    sycl::buffer<T, 1> BufR(&r, sycl::range<1>(1));
+    q.submit([&](sycl::handler &cgh) {
+      auto AccR = BufR.template get_access<sycl::access::mode::write>(cgh);
+      cgh.single_task([=]() {
+        AccR[0] = sycl::ext::oneapi::experimental::native::tanh(T(1.0f));
```

Comment:
I took an approach similar to the tests present in `SYCL/DeviceLib/built-ins` and I generalized it for considering not only `float2` but also all the possible vector sizes.

Considering different values is possible, but will complicate the code in a way that could potentially bring problems since the native functions don't have strict accuracy constraints, i.e., each value could have a different range of approximation depending on the device.

Reply:
Sure, but it is one single range. So, it will be easier to tweak in case of problems. If every value is different up to 16 ranges need to be defined.

If this is necessary, I can modify the code.

Reply:
Testing only the case when all elements of the `vec` have the same value seems like a weak test for this feature.  I think it should be expanded to test the case when each element has a different value.